flowise-embed/src/components/Bot.tsx

```diff
@@ -1045,76 +1045,89 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
 
     setMessages((prevMessages) => [...prevMessages, { message: `${file.name}`, type: 'userMessage', fileUploads: urls }]);
 
-    const checklistPrompt = `CHECKLIST\n${fileMap.checklist}\n\nPlain-text: ${textContent}\n\njson: `;
-    const result = await sendBackgroundMessage(checklistPrompt, urls);
-    try {
-      let jsonData = JSON.parse(result.text);
-      jsonData = sanitizeJson(jsonData);
+    const extractChecklist = async () => {
+      let attempt = 0;
+      let result;
 
-      if (Object.keys(jsonData).includes('error') && Object.keys(jsonData).length === 1) {
-        throw new Error(jsonData.error);
-      }
+      while (attempt < 3) {
+        try {
+          attempt++;
```

Comment:
Não que faria muita diferença, mas você poderia economizar uma ou duas linhas usando um `for` ao invés do `while`, não?